packages/application/src/shell.ts

```diff
@@ -72,7 +72,22 @@ class ApplicationShell extends Widget {
     let topPanel = this._topPanel = new Panel();
     let hboxPanel = this._hboxPanel = new BoxPanel();
     let dockPanel = this._dockPanel = new DockPanel();
-    MessageLoop.installMessageHook(dockPanel, Private.activityClassHook);
+    let dockChildHook = this._dockChildHook = (handler, msg) => {
```

Comment:
You can initialize this in the body of the class, rather than in the constructor. That will make it look more like a method:

```typescript
class Foo {
  constructor() {
    doStuff(this._myMethod);
  }
  private _myMethod = (thing: Thing, other: Other) => {
    ...
  };
}
```

Reply:
Are you saying you'd like it better down there, or just pointing out it could be done? I did both, and thought it looked a little weird among the single-line private attributes below, and also thought it was a bit confusing because it did look like a method then. I'm not set on either way, though, so if you guys like it better the way, I'm happy to change it.

Reply:
For this use case, where it's being used as method rather than a property that can be changed, I prefer it to look like a method. My convention is to put it after all private "formal methods" but before the private property declarations:

https://github.com/phosphorjs/phosphor/blob/master/packages/widgets/src/scrollbar.ts#L598